### DISCUSSION TITLE
ItemAssemblies and New Submarine file clutter in LocalMods

### DISCUSSION THREAD

**Discussion by HalfdeadKiller** [FakeFishGames/Barotrauma#9075] [Ideas and suggestions]
- [X] I have searched the issue tracker to check if the issue has already been reported.

**Description**
Basically every submarine, outpostmodule, itemassembly, etc is now saved as an individual local mod. In the past all submarines went into the Submarines folder, and you then transferred it manually into a localmod for publishing to the workshop. As of now keeping multiple item assemblies and submarines organized is painful, especially in the mod settings menu.

I'm not sure if a more convenient system can be come up with that also allows easy workshop uploading. Perhaps a text box when saving a submarine to tell the game to save the sub into an existing LocalMods folder? This would help keep multiple development versions of outpost modules or submarines in one folder.

I'm not sure either about ItemAssemblies. I used quite a few custom ItemAssemblies but now I either fill up my Enabled Regular Packages with a bunch of ItemAssembly mods, which will get hard to sift through if I want to host a server through the game client with gameplay mods. Or I just don't use my custom item assemblies because they are all individual mods.

Or is it intended that the user goes in, copies all their most used item assemblies into a LocalMod, edits the filelist.xml for each one, and then calls it good? It seems like a lot of extra, potentially unnecessary, work to me, however I'm not familiar with the back end of things.

**Steps To Reproduce**
Save ItemAssemblies or Submarines, and cry internally as Enabled/Disabled Regular Packages starts piling up with stuff.

**Version**
Windows Barotrauma 0.17.15.0

**Additional information**
Here is a screenshot of my current packages menu. I only have like 7 actual gameplay mods, the rest I'm not really sure of anymore because its a pain to sift through it all.
![image](https://user-images.githubusercontent.com/16188887/167089707-c82b8403-37a0-4037-a2f3-57c4a4f6afe1.png)

**Comment by JoneKone** [comment]
I agree.. this is not very optimal, a jump to the correct direction, but not optimal.. And difficult at best for Dedicated server owners. I currently run a script that keeps the old submarine folder alive on my server.

**Comment by HalfdeadKiller** [comment]
It seems I was unaware of Github Discussions, sorry about that.

**Reply by JoneKone** [reply]
Well it's more of a feature request anyways not an real issue.
